jetbrains-core/src/software/aws/toolkits/jetbrains/services/dynamo/editor/DynamoTableEditorProvider.kt

```diff
@@ -0,0 +1,45 @@
+// Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.dynamo.editor
+
+import com.intellij.openapi.fileEditor.FileEditor
+import com.intellij.openapi.fileEditor.FileEditorManager
+import com.intellij.openapi.fileEditor.FileEditorPolicy
+import com.intellij.openapi.fileEditor.FileEditorProvider
+import com.intellij.openapi.fileEditor.OpenFileDescriptor
+import com.intellij.openapi.project.DumbAware
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.vfs.VirtualFile
+import software.aws.toolkits.jetbrains.core.awsClient
+import software.aws.toolkits.jetbrains.utils.notifyError
+import software.aws.toolkits.resources.message
+import software.aws.toolkits.telemetry.DynamodbTelemetry
+import software.aws.toolkits.telemetry.Result
+
+class DynamoTableEditorProvider : FileEditorProvider, DumbAware {
+    override fun accept(project: Project, file: VirtualFile): Boolean = file is DynamoVirtualFile
+
+    override fun createEditor(project: Project, file: VirtualFile): FileEditor = DynamoTableEditor(file as DynamoVirtualFile)
+
+    override fun getEditorTypeId(): String = "DynamoTableEditor"
+
+    override fun getPolicy(): FileEditorPolicy = FileEditorPolicy.HIDE_DEFAULT_EDITOR
+
+    companion object {
+        fun openViewer(project: Project, tableArn: String) {
+            try {
+                val virtualFile = DynamoVirtualFile(tableArn, project.awsClient())
+                FileEditorManager.getInstance(project).openTextEditor(
+                    OpenFileDescriptor(project, virtualFile),
+                    /*focusEditor*/ true
```

Comment:
nit: Are we going to start doing this more vs just having the IDE annotate it dynamically?

Reply:
Im not sure TBH...I wish it was kotlin API so I could use named params. It is hard to know what it is in a PR without it...